diff --git a/io_uring/io_uring.c b/io_uring/io_uring.c
index f170f72125b2..e563ac58c49c 100644
--- a/io_uring/io_uring.c
+++ b/io_uring/io_uring.c
@@ -154,7 +154,7 @@ static __read_mostly DEFINE_STATIC_KEY_FALSE(io_key_has_sqarray);
 struct kmem_cache *req_cachep;
 static struct workqueue_struct *iou_wq __ro_after_init;
 
-static int __read_mostly sysctl_io_uring_disabled = 1;
+static int __read_mostly sysctl_io_uring_disabled;
 static int __read_mostly sysctl_io_uring_group = -1;
 
 #ifdef CONFIG_SYSCTL
@@ -164,9 +164,9 @@ static const struct ctl_table kernel_io_uring_disabled_table[] = {
 		.data		= &sysctl_io_uring_disabled,
 		.maxlen		= sizeof(sysctl_io_uring_disabled),
 		.mode		= 0644,
-		/* only handle a transition from default "1" to "2" */
+		/* disallow decreasing value */
 		.proc_handler	= proc_dointvec_minmax,
-		.extra1		= SYSCTL_TWO,
+		.extra1		= &sysctl_io_uring_disabled,
 		.extra2		= SYSCTL_TWO,
 	},
 	{
