diff --git a/arch/x86/entry/vdso/vdso32/Makefile b/arch/x86/entry/vdso/vdso32/Makefile
index add6afb484ba4..42acca0368108 100644
--- a/arch/x86/entry/vdso/vdso32/Makefile
+++ b/arch/x86/entry/vdso/vdso32/Makefile
@@ -13,7 +13,7 @@ vobjs-y			+= system_call.o sigreturn.o
 # Compilation flags
 flags-y			:= -DBUILD_VDSO32 -m32 -mregparm=0
 flags-$(CONFIG_X86_64)	+= -include $(src)/fake_32bit_build.h
-flags-remove-y          := -m64
+flags-remove-y          := -m64 -mllvm=--enable-ipra
 
 # The location of this include matters!
 include $(src)/../common/Makefile.include
